command/cmd_copy.go

```diff
@@ -153,103 +147,20 @@ func Copy(
 		}
 
 		if err := object.Err; err != nil {
-			printError(fullCommand, op, err)
+			printError(c.fullCommand, c.op, err)
 			continue
 		}
 
 		src := object.URL
-
-		shouldOverrideFunc := func(dst *objurl.ObjectURL) error {
-			return shouldOverride(
-				ctx,
-				src,
-				dst,
-				noClobber,
-				ifSizeDiffer,
-				ifSourceNewer,
-			)
-		}
-
 		var task parallel.Task
 
 		switch {
 		case srcurl.Type == dsturl.Type: // local->local or remote->remote
-			task = func() error {
-				dsturl, err := prepareCopyDestination(ctx, srcurl, src, dsturl, parents)
-				if err != nil {
-					return err
-				}
-
-				err = doCopy(
-					ctx,
-					src,
-					dsturl,
-					op,
-					deleteSource,
-					shouldOverrideFunc,
-					// flags
-					storageClass,
-				)
-				if err != nil {
-					return &errorpkg.Error{
-						Op:  op,
-						Src: src,
-						Dst: dsturl,
-						Err: err,
-					}
-				}
-				return nil
-			}
+			task = c.copy(ctx, srcurl, src, dsturl)
```

Comment:
Copy -> c.copy -> c.doCopy feels like too many redirections to me. If we are going in this direction, we should rename `c.copy` to `c.prepareCopyTask` or similar. Same for `upload` and `download`.

Reply:
done